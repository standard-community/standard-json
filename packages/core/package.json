{
  "name": "@standard-community/standard-json",
  "version": "0.2.0",
  "type": "module",
  "license": "MIT",
  "main": "dist/index.cjs",
  "module": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "pkgroll --minify --clean-dist"
  },
  "keywords": [
    "standard-schema",
    "standard-community",
    "standard-schema-community",
    "json-schema",
    "convertor"
  ],
  "homepage": "https://github.com/standard-community",
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/standard-community/standard-json.git",
    "directory": "packages/core"
  },
  "bugs": {
    "url": "https://github.com/standard-community/standard-json/issues"
  },
  "exports": {
    "import": {
      "types": "./dist/index.d.ts",
      "default": "./dist/index.js"
    },
    "require": {
      "types": "./dist/index.d.cts",
      "default": "./dist/index.cjs"
    }
  },
  "peerDependencies": {
    "@valibot/to-json-schema": "^1.0.0-rc.0",
    "effect": "^3.13.10",
    "zod-to-json-schema": "^3.24.1"
  },
  "peerDependenciesMeta": {
    "@valibot/to-json-schema": {
      "optional": true
    },
    "effect": {
      "optional": true
    },
    "zod-to-json-schema": {
      "optional": true
    }
  },
  "devDependencies": {
    "@standard-schema/spec": "^1.0.0",
    "@types/json-schema": "^7.0.15",
    "arktype": "^2.0.4",
    "pkgroll": "^2.5.1",
    "valibot": "^1.0.0-rc.4",
    "zod": "^3.24.2"
  }
}