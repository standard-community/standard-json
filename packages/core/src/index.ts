import type { StandardSchemaV1 } from "@standard-schema/spec";
import type { JSONSchema7 } from "json-schema";

import { getToJsonSchemaFn } from "./vendors/index.js";

/**
 * Converts a Standard Schema to a JSON schema.
 */
export const toJsonSchema = async (
  schema: StandardSchemaV1,
): Promise<JSONSchema7> =>
  getToJsonSchemaFn(schema["~standard"].vendor).then(async (toJsonSchema) =>
    toJsonSchema(schema),
  );
