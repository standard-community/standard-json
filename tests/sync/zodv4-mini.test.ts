import { describe, expect, it } from "vitest";
import { type $ZodType, toJSONSchema } from "zod/v4/core";
import { z } from "zod/v4/mini";
import { loadVendor, toJsonSchema } from "~/index.js";
import type { ToJsonSchemaFn } from "~/vendors/utils.js";

describe("sync check", () => {
  it("zodv4-mini", () => {
    // Loading the vendor
    loadVendor("zod", ((schema, options) =>
      toJSONSchema(schema as $ZodType, options)) as ToJsonSchemaFn);

    const zod4Schema = z.object({
      myString: z.string(),
      myUnion: z.union([z.number(), z.boolean()]),
    });

    const zod4JsonSchema = toJsonSchema(zod4Schema);
    expect(zod4JsonSchema).toMatchSnapshot();
  });
});
