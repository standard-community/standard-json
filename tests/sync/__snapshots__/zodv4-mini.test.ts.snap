// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`sync check > zodv4-mini 1`] = `
{
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "additionalProperties": false,
  "properties": {
    "myString": {
      "type": "string",
    },
    "myUnion": {
      "anyOf": [
        {
          "type": "number",
        },
        {
          "type": "boolean",
        },
      ],
    },
  },
  "required": [
    "myString",
    "myUnion",
  ],
  "type": "object",
}
`;
